channeldb/invoices.go

```diff
@@ -1185,6 +1149,13 @@ func updateInvoice(invoices, settleIndex *bbolt.Bucket, invoiceNum []byte,
 	if preUpdateState != invoice.Terms.State &&
 		invoice.Terms.State == ContractSettled {
 
+		if invoice.Terms.PaymentPreimage == UnknownPreimage {
```

Comment:
Wasn't this checked elsewhere before? This strikes me more as something that should happen in the registry rather than as a database modification constraint.

Reply:
Yes, I did it here because I didn't want to set a preimage in the regular invoice settle path that was just copied from the invoice itself.

I looked at it again because you bring it up here and I indeed think it is better to keep all of it in the invoice registry. Fixed.